sdk/core/core-lro/src/poller.ts

```diff
@@ -243,9 +245,9 @@ export abstract class Poller<TState extends PollOperationState<TResult>, TResult
    */
   constructor(operation: PollOperation<TState, TResult>) {
     this.operation = operation;
-    this.promise = new Promise(
+    this.promise = new Promise<TResult>(
```

Comment:
If we change this to `TResult | undefined` we probably won't need to change that question mark below.

Reply:
Thanks @sadasant for the suggestion! I opted to use a different approach here: https://github.com/Azure/azure-sdk-for-js/pull/12770#discussion_r542510189.